src/crystal/system/win32/time.cr

```diff
@@ -163,7 +164,23 @@ module Crystal::System::Time
   # Searches the registry for an English name of a time zone named *stdname* or *dstname*
   # and returns the English name.
   private def self.translate_zone_name(stdname, dstname)
-    # TODO: Needs implementation once there is access to the registry.
-    nil
+    Registry::LOCAL_MACHINE.open("SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Time Zones",
+      Registry::SAM::ENUMERATE_SUB_KEYS | Registry::SAM::QUERY_VALUE) do |key|
+      key.each_key do |sub_name|
+        key.open(sub_name, Registry::SAM::READ) do |sub_key|
```

Comment:
I _think_ you can write it as `key.open(sub_name, :read)`, right?

Reply:
(no need to change it, though, I was just curious)

Reply:
I think I had actually implemented that part before #6074 was merged.

Looking at it, I figured the argument to the previous call didn't need to be a combination of *enumerate* and *query*. Changing this all to `:read` makes the code a little bit simpler.